lib/NavigatorWatcher.js

```diff
@@ -40,9 +40,12 @@ class NavigatorWatcher {
 
     this._eventListeners = [];
 
-    const watchdog = new Promise(fulfill => this._maximumTimer = setTimeout(fulfill, this._timeout))
-        .then(() => 'Navigation Timeout Exceeded: ' + this._timeout + 'ms exceeded');
-    const navigationPromises = [watchdog];
+    const navigationPromises = [];
+    if (this._timeout) {
+      const watchdog = new Promise(fulfill => this._maximumTimer = setTimeout(fulfill, this._timeout))
+          .then(() => 'Navigation Timeout Exceeded: ' + this._timeout + 'ms exceeded');
+      navigationPromises.push(watchdog);
+    }
```

Comment:
this is as the @aslushnikov suggestion, it's simple and clear code. But it think it still have problem if user pass timeout 0 as a string.

Reply:
@brutalcrozt yes, we should rely on documentation in this case and expect user to pass in a number. There's also a check that timeout is always a number: https://github.com/GoogleChrome/puppeteer/blob/c46c41d89caefd1e1951055fb041521a2bcc933b/lib/NavigatorWatcher.js#L28